astroquery/mast/cutouts.py

```
@@ -637,3 +637,163 @@ def get_cutouts(self, coordinates, *, size=5, survey=None):
+
+        Parameters
+        ----------
+        coordinates : str or `astropy.coordinates` object
+            The target around which to search. It may be specified as a
+            string or as the appropriate `astropy.coordinates` object.
+        size : int, array-like, `~astropy.units.Quantity`
+            Optional, default 5 pixels.
+            The size of the cutout array. If ``size`` is a scalar number or
+            a scalar `~astropy.units.Quantity`, then a square cutout of ``size``
+            will be created.  If ``size`` has two elements, they should be in
+            ``(ny, nx)`` order.  Scalar numbers in ``size`` are assumed to be in
+            units of pixels. `~astropy.units.Quantity` objects must be in pixel or
+            angular units.
+
+        Returns
+        -------
+        response : A list of `~astropy.io.fits.HDUList` objects.
+        """
+
+        # Get Skycoord object for coordinates/object
+        coordinates = parse_input_location(coordinates)
+
+        param_dict = _parse_cutout_size(size)
+
+        # Need to convert integers from numpy dtypes
+        # so we can convert the dictionary into a JSON object
+        # in service_request_async(...)
+        param_dict["x"] = int(param_dict["x"])
+        param_dict["y"] = int(param_dict["y"])
+
+        # Adding RA and DEC to parameters dictionary
+        param_dict["ra"] = coordinates.ra.deg
+        param_dict["dec"] = coordinates.dec.deg
+
+        response = self._service_api_connection.service_request_async("astrocut", 
+                                                                      param_dict, 
+                                                                      use_json=True)
+        response.raise_for_status()  # Raise any errors
+
+        try:
+            ZIPFILE = zipfile.ZipFile(BytesIO(response.content), 'r')
+        except zipfile.BadZipFile:
+            message = response.json()
+            warnings.warn(message['msg'], NoResultsWarning)
```

I think we should distinguish between an error in the response as opposed to a null response. I agree that it should not be an exception if there are no results, but if the response contains a genuine error message, we should raise that rather than warn about it.

---

I made the distinction by way of an `if`/`else` statement. Let me know if this is what you were thinking of.